### PR TITLE
auto-update documented version for external projects

### LINKED ISSUE #1128: Create automation script to update project version numbers across repos.
Throughout this repo we reference specific version numbers for things like Go-IPFS and IPFS Updater.

image

The problem is that where there's updates to these applications, we have to manually go through this repo and update any references. It'd be nice if we had a script that:

1. Knew which pages to look at.
2. Found version numbers in those pages.
3. Checked to see if they were the latest version of the relevant apps/programs.
4. Updates version numbers where necessary.

We can use this issue to discuss the best way of creating this script.

### PR DESCRIPTION
**Pull request by laurentsenta**
Fixes #1128

- Re-use the documentation generation we used for go-ipfs with ipfs-udate and ipfs-cluster,
- Remove the concept of tags when creating PR, etc (because we're not tied to a single go-ipfs version anymore),
- Make the workflow callable.

## Todo

- [x] add version bump for ipfs-cluster and ipfs-update
- [x] update image to use `jq`
- [x] rename and re-organize the workflow for clarity
- [x] tweak the action so it always uses the same branch
- [x] test base case (updating missing version, etc).
- [x] test the complex case (actual IPFS update) still works
- [x] check the content of the latest repo-sync/pull-request

### CODE REVIEW

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

**Comment by laurentsenta** [comment]
Testing:

- first run (outdated dependencies) https://github.com/laurentsenta/ipfs-docs/runs/6757005580?check_suite_focus=true
- PR: https://github.com/laurentsenta/ipfs-docs/pull/1
- second run (pr was still open) https://github.com/laurentsenta/ipfs-docs/actions/runs/2448332670
- third run (after merging the PR) https://github.com/laurentsenta/ipfs-docs/actions/runs/2448343640

File: .github/actions/update-with-latest-versions/entrypoint.sh
Comment (laurentsenta) [inline .github/actions/update-with-latest-versions/entrypoint.sh:42]: Note: github doesn't detect the move so it looks like I created the whole file. This PR actually mainly move the update version code lower and calls it three times.

- We also push to a unique branch on file changes, instead of something like `new-documentation-${NEW_TAG}` because documentation updates are not only tied to go-ipfs versions anymore.

- I was thinking about extracting the version replace script to its own action, but I'm under the impression that would make the code more complex (spread it around 3 actions instead of a single script file).

File: .github/workflows/update-on-new-ipfs-tag.yml
Comment (laurentsenta) [inline .github/workflows/update-on-new-ipfs-tag.yml:23]: I started seeing a workfspace error (https://github.com/laurentsenta/ipfs-docs/runs/6756806053?check_suite_focus=true) in my github action, had to update. I think the previous version was a typo (2.6.1, not 16)

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

**Review by lidel — APPROVED** [review]
Thank you @laurentsenta :heart: 

LGTM (`/releases/latest` skips release candidates, and only shows final releases, which is what we want).

Let's merge this (if there are any issues, we can improve in follow-up PR)

File: .github/actions/update-with-latest-versions/entrypoint.sh
Comment (lidel) [inline .github/actions/update-with-latest-versions/entrypoint.sh:67]: @johnnymatthews for now we track updates of these three, but if we ever add more projects, we can add them here with ease (as long they are on github and publish releases there)

**Pull request merged**
